submissions/ses3332002/memory-game/script.js

```diff
@@ -0,0 +1,140 @@
+"use strict";
+  
+  const doc = document.body;
+  const themeButtons = document.querySelector(".theme_selector");
+  const sizeSelector = document.querySelector(".size_selector");
+  const cardThemeSelector = document.querySelectorAll('input[name="option"]');
+  const startButton = document.querySelector(".start_button");
+  const gameArea = document.querySelector(".game_area");
+  const switchableControls = document.querySelectorAll(".switchable");
+  let numbersArray;
+  let cardThemeSelected;
+  let cardSelected;
+  let cardsArray;
+  let matchedCounter;
+  
+  startButton.addEventListener("click", startGame);
+  doc.classList.add("theme1");
+  
+  themeButtons.addEventListener("click", function({target}) {
+    if (target.tagName != "BUTTON") {
+      return;
+    };
+    doc.classList.remove(doc.classList[0]);
+    doc.classList.add(target.dataset.theme);
+  });
+  
+  function startGame() {
+    cardThemeSelected = document.querySelector('input[name="option"]:checked');
+    matchedCounter = 0;
+    prepareGameField();
+    switchControls();
+  };
+  
+  function switchControls() {
+    switchableControls.forEach((item) => {
+      item.disabled = !item.disabled;
+      item.classList.toggle("form_item-desabled");
+    });
+  };
+  
+  function prepareGameField() {
+    numbersArray = [];
```

Comment:
You can define this variable as an empty array at the beginning of your code. It'll help you not to make reassigning here.

Reply:
Well, in one "game session" there can be several games with different gamefield sizes, so player can press "Start" button several times, so I need to clear old numbersArray anyway while I prepare new gamefield in prepareGameField() function.